apps_product_creator/models/odoo_module.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2017-Today: Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class OdooModule(models.Model):
+    _inherit = "odoo.module"
+
+    @api.depends("product_template_id", "product_template_id.product_variant_ids")
+    def _compute_product_qty(self):
+        for module in self:
+            module.product_qty = len(module.product_template_id.product_variant_ids)
+
+    product_template_id = fields.Many2one("product.template", "Product Template",)
+    product_qty = fields.Integer(
+        "# of Products", compute="_compute_product_qty", store=True
+    )
+
+    def action_view_products(self):
+        action = self.env.ref("product.product_normal_action_sell")
+        result = action.read()[0]
+        product_ids = sum(
+            [m.product_template_id.product_variant_ids.ids for m in self], []
+        )
+        # choose the view_mode accordingly
+        if len(product_ids) > 1:
+            result["domain"] = "[('id','in',[" + ",".join(map(str, product_ids)) + "])]"
```

Comment:
I think that is not really necessary represent the id as a string
```suggestion
            result["domain"] = "[('id','in', %s)]" % str(product_ids)
```